src/Storage/Collection/Relations.php

```diff
@@ -234,13 +236,39 @@ public function offsetGet($offset)
         return $collection;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function serialize()
     {
-        $output = [];
-        foreach ($this as $k => $existing) {
-            $output[$existing->getToContenttype()][] = spl_object_hash($existing);
+        $trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 1);
+        if (isset($trace[0]['file'])) {
+            Deprecated::method(3.4, 'toArray');
+
+            $output = [];
+            foreach ($this as $k => $existing) {
+                $output[$existing->getToContenttype()][] = spl_object_hash($existing);
+            }
+
+            return $output;
+        }
+
+        $data = [];
+        foreach ($this as $element) {
+            $data[] = $element;
         }
 
-        return $output;
+        return serialize($data);
+    }
```

Comment:
So basically [`\Bolt\Storage\Entity\MagicAttributeTrait::__call()`](https://github.com/bolt/bolt/blob/release/3.4/src/Storage/Entity/MagicAttributeTrait.php#L76-L80) is catching method called, or prefixed with, "serialize" …